nsot/models.py

```diff
@@ -1470,6 +1470,105 @@ def to_dict(self):
         }
 
 
+class Circuit(Resource):
+    """Represents two network Interfaces that are connected"""
+
+    # a_endpoint interface
+    a_endpoint = models.ForeignKey(
+        Interface, db_index=True, related_name='circuits_a',
+        null=False, verbose_name='A-side Interface',
```

Comment:
Well, it's the other way around. `a_endpoint` should be required but `z_endpoint` MAY be a third-party if the device is on the edge and it's a peering/transit/PNI. Having a `z_endpoint` in that case would require you adding a device and interface for the third party which is kinda unrealistic/unneeded IMO.